frontend/src/reducers/plannerSlice.js

```diff
@@ -0,0 +1,292 @@
+      }
+      localStorage.setItem("planner", JSON.stringify(state));
+    },
+    removeCourse: (state, action) => {
+      // Remove courses from years and courses
+      if (state.courses[action.payload]) {
+        const { plannedFor } = state.courses[action.payload];
+
+        // remove course
+        delete state.courses[action.payload];
+
+        if (plannedFor) {
+          // course must already been planned
+          // Example plannedFor: '2021t2'
+          const yearIndex = parseInt(plannedFor.slice(0, 4), 10) - state.startYear;
+          const term = plannedFor.slice(4);
+          // remove the course from the year and term
+          state.years[yearIndex][term] = state.years[yearIndex][term].filter(
+            (course) => course !== action.payload,
+          );
+        } else {
+          // course must be in unplanned
+          state.unplanned = state.unplanned.filter(
+            (course) => course !== action.payload,
+          );
+        }
+        localStorage.setItem("planner", JSON.stringify(state));
+      }
+    },
+    removeAllCourses: (state) => {
+      state.years = generateEmptyYears(state.numYears);
+      state.courses = {};
+      state.unplanned = [];
+      localStorage.setItem("planner", JSON.stringify(state));
+    },
+    unschedule: (state, action) => {
+      const code = action.payload;
+      state.unplanned.push(code);
+
+      const { plannedFor } = state.courses[code];
+
+      const yearI = parseInt(plannedFor.slice(0, 4), 10) - state.startYear;
+      const termI = plannedFor.slice(4);
+
+      state.years[yearI][termI] = state.years[yearI][termI].filter((course) => course !== code);
+
+      state.courses[code].plannedFor = null;
+      state.courses[code].isUnlocked = true;
+      state.courses[code].warnings = [];
+      state.courses[code].handbookNote = "";
+      state.courses[code].isAccurate = true;
+
+      localStorage.setItem("planner", JSON.stringify(state));
+    },
+    unscheduleAll: (state) => {
+      Object.entries(state.courses).forEach(([code, desc]) => {
+        if (desc.plannedFor !== null) {
+          state.unplanned.push(code);
+
+          const { plannedFor } = state.courses[code];
+
+          const yearI = parseInt(plannedFor.slice(0, 4), 10) - state.startYear;
+          const termI = plannedFor.slice(4);
+
+          state.years[yearI][termI] = state.years[yearI][termI].filter((course) => course !== code);
+
+          state.courses[code].plannedFor = null;
+          state.courses[code].isUnlocked = true;
+          state.courses[code].warnings = [];
+          state.courses[code].handbookNote = "";
+          state.courses[code].isAccurate = true;
+        }
+      });
+      localStorage.setItem("planner", JSON.stringify(state));
+    },
+    toggleSummer: (state) => {
+      state.isSummerEnabled = !state.isSummerEnabled;
+
+      // TODO: Unsure if we should keep the course in the summer column if hidden
```

Comment:
what is going on here?

Reply:
So there's an edge case that I encountered while testing out the term planner. Let's say that a user has the summer term toggled on when planning their courses. The user drags their courses into the summer column. However, the user decides to not show the summer column anymore and toggles the summer term to not be shown. There's two intents to consider:
- If the user simply does not want to do a summer term in the degree, should all the planned courses in the summer term be moved back to unplanned? The current implementation does not does this and the summer courses are simply left in the summer column (which is hidden). This means that the user can't drap/plan their degree for courses in the summer column without toggling the summer term back on.
- If the user simply wants to hide the summer term, toggle off the summer column and toggle on the summer column after, should we save the summer courses?

I still need to see if there's a better way to handle this but this would be out of the scope of the ticket I feel (and just want to merge the redux changes into dev as soon lol). Personally, I think the first option would be simplier but any thoughts on this is welcomed :D

Reply:
hmm im not too fussed by this - i think its fine to leave the courses in the summer term. as long as we get the dragndrop to unplanned working the difference in the approaches should be minimal.